Shared/messages/MessageFeed.cpp

```diff
@@ -17,44 +17,156 @@
+  }
+  else
+  {
+    // set the entity id field
+    entity_id_field_name = Message::GEOMESSAGE_ID_NAME;
+
+    // set the fields
+    field_names.reserve(11);
+    field_names.emplace_back(Message::GEOMESSAGE_TYPE_NAME);
+    field_names.emplace_back(Message::GEOMESSAGE_ACTION_NAME);
+    field_names.emplace_back(Message::GEOMESSAGE_UNIQUE_DESIGNATION_NAME);
+    field_names.emplace_back(Message::GEOMESSAGE_WKID_NAME);
+    field_names.emplace_back(Message::GEOMESSAGE_SIC_NAME);
+    field_names.emplace_back(Message::GEOMESSAGE_CONTROL_POINTS_NAME);
+    field_names.emplace_back(Message::GEOMESSAGE_UNIQUE_DESIGNATION_NAME);
+    field_names.emplace_back(Message::GEOMESSAGE_STATUS_911_NAME);
+    field_names.emplace_back(Message::GEOMESSAGE_ENVIRONMENT_NAME);
+  }
+  m_entityIdAttributeName = entity_id_field_name;
+  field_names.emplace_back(entity_id_field_name);
+  field_names.emplace_back(Message::SIDC_NAME);
+
+  QList<Field> fields;
+  fields.reserve(field_names.count());
+  for (auto& fn : field_names)
+  {
+    fields.emplace_back(FieldType::Text, fn, "", 256, Domain(), false, true);
+  }
+
+  // build the dynamic entity data source info from the fields and the entity id field name
+  auto* dynamicEntityDataSourceInfo = new DynamicEntityDataSourceInfo(entity_id_field_name, fields, this);
+  dynamicEntityDataSourceInfo->setSpatialReference(SpatialReference::wgs84());
+
+  // listen for new entities
+  connect(this, &DynamicEntityDataSource::dynamicEntityReceived, this, [this](DynamicEntityInfo* info)
+  {
+    // check new entity for select/unselect action
+    auto* dynamicEntity = info->dynamicEntity();
+    dynamicEntity->setParent(this);
+    checkEntityForSelectAction(dynamicEntity);
+
+    // mark the info as delete later so it can be cleaned up
+    // info in the source cannot be cleaned up immediately since alert targets might need a reference to the related DynamicEntity
+    info->deleteLater();
+  });
+
+  // listen for new observations
+  connect(this, &DynamicEntityDataSource::dynamicEntityObservationReceived, this, [this](DynamicEntityObservationInfo* observationInfo)
+  {
+    // check new entity for select/unselect action
+    auto* dynamicEntity = observationInfo->observation()->dynamicEntity();
+    checkEntityForSelectAction(dynamicEntity);
+
+    // mark the observation as delete later so it can be cleaned up
+    // observations in the source cannot be cleaned up immediately since alert targets might need a reference to the related DynamicEntity
+    observationInfo->deleteLater();
+  });
+
+  // remove ownership when entities are purged
+  connect(this, &DynamicEntityDataSource::dynamicEntityPurged, this, [](DynamicEntityInfo* info)
+  {
+    // release the dynamic entity
+    auto* dynamicEntity = info->dynamicEntity();
+    dynamicEntity->setParent(nullptr);
+    dynamicEntity->deleteLater();
+
+    // mark the info as delete later so it can be cleaned up
+    info->deleteLater();
```

Comment:
Nice 👍 

I think you can simplify some of this. No need to `dynamicEntity->setParent(nullptr);` if you're going to `dynamicEntity->deleteLater();` on the next line. I don't think it matters if it has a parent or not if you call `deleteLater` on it.

Reply:
Wasn't sure about this one. Saw mention of it not needing to be set to null in some discussions but could not determine for sure in the Qt docs anywhere.

Removed.